gui.py

```diff
@@ -202,6 +221,7 @@ def create_canvas(self):
             self.canvas = text_area
 
         self.canvas_box.addWidget(self.canvas)
+        self.canvas_box.addWidget(savebutton)
```

Comment:
It's nice button. I would also add an entry in the menu. Just saying, not necessary.